internal/controller/pod_controller.go

```diff
@@ -49,77 +49,78 @@ func (a *PodAdmissionWebhook) InjectDecoder(d *admission.Decoder) error {
 // the proxy sidecars on all workloads to match the AuthProxyWorkload config.
 func (a *PodAdmissionWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
 	l := logf.FromContext(ctx)
-	wl := &workload.PodWorkload{
-		Pod: &corev1.Pod{},
-	}
-	err := a.decoder.Decode(req, wl.Object())
+	p := corev1.Pod{}
+	err := a.decoder.Decode(req, &p)
 	if err != nil {
 		l.Info("/mutate-pod request can't be processed",
 			"kind", req.Kind.Kind, "ns", req.Namespace, "name", req.Name)
 		return admission.Errored(http.StatusInternalServerError, err)
 	}
 
+	updatedPod, err := a.handleCreatePodRequest(ctx, p)
```

Comment:
I split the Handle function into 2 separate responsibilities:
- `func Handle(ctx, req)` does all the transformation between `admission.Request` and `admission.Response` into useful domain objects. 
- `func handleCreatePodRequest(ctx, pod)` has the business logic to decide when and how to add Proxy containers to a pod as it is being created.

This made it a lot easier to write unit tests for the business logic without having to implement a lot of extraneous json handling code.